node/src/actors/chain_manager/mining.rs

```diff
@@ -925,14 +928,22 @@ fn build_block(
             }
         };
 
-        update_utxo_diff(
-            &mut utxo_diff,
-            dr_tx.body.inputs.iter().collect(),
-            dr_tx.body.outputs.iter().collect(),
-            dr_tx.hash(),
-        );
-        data_request_txns.push(dr_tx.clone());
-        transaction_fees += transaction_fee;
+        let new_dr_weight = dr_weight + transaction_weight;
```

Comment:
Ok this is important, it may be possible to create a data request with `weight = u32::MAX` by setting the number of witnesses to `u16::MAX`.

```suggestion
        let new_dr_weight = dr_weight.checked_add(transaction_weight);
```